### DISCUSSION TITLE
vcpkg Binary Caching on Azure DevOps for large packages

### DISCUSSION THREAD

**Discussion by Longhanks** [microsoft/vcpkg#26926] [Q&A]
Hello!

We're trying to use the binary caching feature with Azure DevOps Artifacts using the NuGet backend. However, we're unable to save artifacts of for example qtdeclarative: The nuget package is abou 600 MB in size, but Azure DevOps limits packages to 500 MB.

This error message appears when trying to upload the artifact:

```
Pushing qtdeclarative_x64-windows.6.3.1-vcpkgLONG_NUMBER.nupkg to '[https://pkgs.dev.azure.com/YOUR_ORG/LONG_NUMBER/_packaging/LONG_NUMBER/nuget/v2/'...](https://pkgs.dev.azure.com/YOUR_ORG/LONG_NUMBER/_packagingLONG_NUMBER/nuget/v2/%27...)
  PUT https://pkgs.dev.azure.com/YOUR_ORG/LONG_NUMBER/_packaging/LONG_NUMBER/nuget/v2/
  RequestEntityTooLarge https://pkgs.dev.azure.com/YOUR_ORG/LONG_NUMBER/_packaging/LONG_NUMBER/nuget/v2/ 33946ms
Response status code does not indicate success: 413 (Request Entity Too Large - Package is too large. Size limit is 524288000 bytes.
```

Since it is not possible to split the package in e. g. debug and release builds, I wonder what solution is recommended to store binary artifacts of this size? Azure blob storage is experimental.

Thanks for any suggestion!

**Comment by Osyotr** [comment]
The vcpkg CI experiences the same problem...
From Discord by @BillyONeal:
![image](https://user-images.githubusercontent.com/8740768/191829727-738d8705-354d-4896-b9ee-c692afdac8eb.png)

**Reply by drewhubley** [reply]
I'm having this same issue with Intel MKL libraries
